girder/utility/abstract_assetstore_adapter.py

```diff
@@ -440,3 +440,13 @@ def open(self, file):
         :rtype: FileHandle
         """
         return FileHandle(file, self)
+
+    def getLocalFilePath(self, file):
+        """
+        If an assetstore adapter supports it, return a path to the file on the
+        local file system.
+
+        :param file: The file document.
+        :returns: a local path to the file or None if no such path is known.
+        """
+        return None
```

Comment:
Can this raise an exception? This would be preferable for two reasons:
* The return type could be a single type (`str`). IMO, Girder abuses Python's dynamic typing a bit too much, such that conditional checks are necessary to handle lots of seemingly-simple function outputs.
* In cases where a path is expected to "just work" you can call the method without having to validate the output. In those cases, an uncaught exception would actually be desirable, as it indicates a failure of downstream assumptions.
* In cases where a path may not be available, it allows more information on the cause of the failure to be returned.
* It allows more complex implementations of `getLocalFilePath` (e.g. those with nested function calls) to efficiently abort when they fail.